### DISCUSSION TITLE
v3.0.0

### DISCUSSION THREAD

**Discussion by nickfloyd** [octokit/octokit.net#2567] [Announcements]
Hey friends,

I wanted to highlight this specific deployment for a couple of reasons:

First, because it contained another breaking change.  While the breaking change most likely will not impact many folks, the contract did change on the implementation, and given we version using simver; we bumped the major again.  I realize that for years we were setting at prerelease, and now over a couple of months, we're setting at 3.x.  This was bound to happen, and it was time to move past the pre-release phase of this SDK.  Over time (and heading into generative SDKs), these version boosts will slow. I just wanted to call it out here, given something like this could appear reckless - the reality is that this is us growing into this SDK as the new caretakers while dealing with a long tail of much-needed updates.

Secondly, we're beginning to walk into the ideal of generative APIs (there is a standardization changeset below that is part of all of that).  Please join the discussion (https://github.com/octokit/octokit.net/discussions/2527) around the future of what we have here and how we'd like to get there - right now, these are just baby steps on the way to something awesome; we need you to help us get there!

The details of the release are below...

## Advisories and Breaking Changes

API METHOD CONTRACT CHANGE: In this release, the return type/exception implementation of `AddCollaborator` has changed by fixing a failure with 201 status code when adding a collaborator - [#2559](https://github.com/octokit/octokit.net/pull/2559) via [@annchous](https://github.com/annchous)

RE: Converts response model property accessors from protected to the more appropriate private for consistency - [#2565](https://github.com/octokit/octokit.net/pull/2565) via [@nickfloyd](https://github.com/nickfloyd) - in support of Generative SDKs

### Milestone: We modified the existing source to help unify it and prepare for our generative SDK work.  

**Features/Enhancements**

- Added Packages and Package versions APIs - [#2551](https://github.com/octokit/octokit.net/pull/2551) via [@JonruAlveus](https://github.com/JonruAlveus)
- Reworks how Locking and Unlocking works on Issues and adds Locking and Unlocking to Pull Requests - [#2543](https://github.com/octokit/octokit.net/pull/2543) via [@notauserx](https://github.com/notauserx)
- Adds the ability to pass ApiOptions to PullRequestsClient.Files - [#2553](https://github.com/octokit/octokit.net/pull/2553) via [@Gamer025](https://github.com/Gamer025)
- Added support for generating release notes.  😉 - [#2563](https://github.com/octokit/octokit.net/pull/2563) via [@Cristaling](https://github.com/Cristaling)
- Adds missing fields for deployments API - [#2560](https://github.com/octokit/octokit.net/pull/2560) via [@hansmbakker](https://github.com/hansmbakker), [@nickfloyd](https://github.com/nickfloyd)
- Adds credential-authorization get endpoints - [#2556](https://github.com/octokit/octokit.net/pull/2556) via [@JonruAlveus](https://github.com/JonruAlveus), [@nickfloyd](https://github.com/nickfloyd)

**Fixes**

- Fix failure with 201 status code on adding a collaborator - [#2559](https://github.com/octokit/octokit.net/pull/2559) via [@annchous](https://github.com/annchous)

**Housekeeping**

- Bumps [Microsoft.NET.Test.Sdk](https://github.com/microsoft/vstest) from 17.2.0 to 17.3.0. - [#2548](https://github.com/octokit/octokit.net/pull/2548) via [@dependabot[bot]](https://github.com/apps/dependabot)
- Bumps [Microsoft.NET.Test.Sdk](https://github.com/microsoft/vstest) from 17.3.0 to 17.3.1. - [#2561](https://github.com/octokit/octokit.net/pull/2561) via [@dependabot[bot]](https://github.com/apps/dependabot)
- Bumps Microsoft.NETFramework.ReferenceAssemblies from 1.0.2 to 1.0.3. - [#2549](https://github.com/octokit/octokit.net/pull/2549) via [@dependabot[bot]](https://github.com/apps/dependabot)

**Documentation Updates**

- Removed reference to Basic Authentication in the docs - [#2555](https://github.com/octokit/octokit.net/pull/2555) via [@rkreisel](https://github.com/rkreisel)

**Full Changelog**: https://github.com/octokit/octokit.net/compare/v2.0.1...v3.0.0

<hr /><em>This discussion was created from the release <a href='https://github.com/octokit/octokit.net/releases/tag/v3.0.0'>v3.0.0</a>.</em>

**Comment by CalvinWilkinson** [comment]
Managing versions and how they can change can be difficult.  Completely understand where you are coming from.  Sometimes it cannot be helped but the fact that it is communicated is the important part.  😀

Thank's for the update!!
